test/integration/esys-macsequencestart.int.c

```diff
@@ -0,0 +1,289 @@
+    inPublic.publicArea.objectAttributes |= TPMA_OBJECT_SIGN_ENCRYPT;
+    inPublic.publicArea.objectAttributes |= TPMA_OBJECT_USERWITHAUTH;
+    inPublic.publicArea.objectAttributes |= TPMA_OBJECT_SENSITIVEDATAORIGIN;
+    inPublic.publicArea.parameters.keyedHashDetail.scheme.scheme = TPM2_ALG_HMAC;
+    inPublic.publicArea.parameters.keyedHashDetail.scheme.details.hmac.hashAlg = TPM2_ALG_SHA256;
+
+    TPM2B_AUTH auth = {.size = 20,
+                       .buffer={10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+                                20, 21, 22, 23, 24, 25, 26, 27, 28, 29}};
+
+    r = Esys_CreatePrimary(esys_context, ESYS_TR_RH_OWNER, ESYS_TR_PASSWORD,
+                           ESYS_TR_NONE, ESYS_TR_NONE, &inSensitivePrimary,
+                           &inPublic, &outsideInfo, &creationPCR,
+                           &primaryHandle, &outPublic, &creationData,
+                           &creationHash, &creationTicket);
+    goto_if_error(r, "Error: CreatePrimary", error);
+
+    r = Esys_TR_SetAuth(esys_context, primaryHandle, &authValuePrimary);
+    goto_if_error(r, "Error: TR_SetAuth", error);
+
+    TPMI_ALG_HASH hashAlg = TPM2_ALG_SHA256;
+    ESYS_TR sequenceHandle;
+
+    r = Esys_MAC_Start(esys_context,
+                        primaryHandle,
+#ifdef TEST_SESSION
+                        session,
+#else
+                        ESYS_TR_PASSWORD,
+#endif
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        &auth,
+                        hashAlg,
+                        &sequenceHandle
+                        );
+    goto_if_error(r, "Error: HashSequenceStart", error);
+
+    TPM2B_MAX_BUFFER buffer ={.size = 20,
+                              .buffer={10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+                                       20, 21, 22, 23, 24, 25, 26, 27, 28, 29}};
+    r = Esys_TR_SetAuth(esys_context, sequenceHandle, &auth);
+    goto_if_error(r, "Error esys TR_SetAuth ", error);
+
+    r = Esys_SequenceUpdate(esys_context,
+                            sequenceHandle,
+#ifdef TEST_SESSION
+                            session,
+#else
+                            ESYS_TR_PASSWORD,
+#endif
+                            ESYS_TR_NONE,
+                            ESYS_TR_NONE,
+                            &buffer
+                            );
+    goto_if_error(r, "Error: SequenceUpdate", error);
+
+    r = Esys_SequenceComplete(esys_context,
+                              sequenceHandle,
+#ifdef TEST_SESSION
+                              session,
+#else
+                              ESYS_TR_PASSWORD,
+#endif
+                              ESYS_TR_NONE,
+                              ESYS_TR_NONE,
+                              &buffer,
+                              ESYS_TR_RH_OWNER,
+                              &result,
+                              &validation
+                              );
+    goto_if_error(r, "Error: SequenceComplete", error);
+
```

Comment:
If you want to verify the HMAC, you do it with ESys_VerifySignature. MAC's are a "symmetric signature".

Reply:
We have other tests fot this.